### ISSUE TITLE
update channel error Resource not found NotFoundException: Resource not found when amplify add notifications

### ISSUE THREAD

**Issue by marcojr** [aws-amplify/amplify-cli#5056]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
Error message: update channel error
Resource not found

**Amplify CLI Version**
4.27.0

**To Reproduce**
- execute amplify add notifications
- Choose APNS or APNS Sandbox
- This will generate with no issues
- Now, execute the same step, but choose FCM
- You will be prompted with ApiKey
- Copy the key from Firebase Console / Project Overview / Gearbox Icon / Project Settings Menu Options / Cloud Messaging Tab / Project Credentials / Server Key (exactly like the official doc asks)

**Expected behavior**
Settings to be stored at <projectDir>/amplify/*

**Screenshots**
![Screenshot 2020-08-11 at 18 19 59](https://user-images.githubusercontent.com/856617/89928322-90c20800-dbff-11ea-9f4a-9672377d50fc.png)

**Desktop (please complete the following information):**
 - OS: MAC
 - Node Version. v12.18.2

**Comment by jcbdev** [comment]
I have this exact same issue

**Comment by jcbdev** [comment]
@SwaySway I have lost a week to issues with notification and analytics since trying to add it to an existing project.  It is totally broken at the moment (see my other issue too - https://github.com/aws-amplify/amplify-cli/issues/5063).  I have managed to hack around it by destroying and recreating environments and backend folders from existing pulls but there is something fundamentally broken after analytics is added to a project.

I really cannot see how I would get this into production at the moment.

**Comment by attilah** [comment]
@marcojr @jcbdev I tried to reproduce this issue with 4.27.2 (the latest version as of now) and I was not able to do it. Both notifications were successfully added. Try with the latest version and get back to me with the results.

Based on the stacktrace that error happens when Amplify CLI tries to retrieve a given pinpoint resource but cannot succeed. Are you using admin credentials or a scoped down user when executing the CLI commands?

**Comment by jcbdev** [comment]
@attilah I think I was able to fix it my end and it's potentially due to the other bug I am having with notifications/analytics.

I had added analytics but it failed for a different reason (see other issue) so I removed analytics by switching envs (created a "devb", switched, destroyed "dev", recreated "dev", switched back, removed "devb").  This was the only way I could get back to a deployable version. The other issue is this - https://github.com/aws-amplify/amplify-cli/issues/5063

Anyway, I noticed yesterday that after re-adding analytics when I got around my other issue that when it gets added the "appId" for pinpoint does not change from the previous values (the original attempt to add them) in the following files: parameters.json and nest-cloudformation-stack.yml

If I update these files with the correct new appId from the new stack it works.  There seems to be a bug where if the values already exist it does not update them.
